src/js/content_scripts/gamepad-navigator.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright (c) 2020 The Gamepad Navigator Authors
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* eslint-env browser */
+
+(function (fluid) {
+    "use strict";
+
+    var gamepad = fluid.registerNamespace("gamepad");
+
+    fluid.defaults("gamepad.gamepadNavigator", {
+        gradeNames: ["fluid.modelComponent"],
+        model: {
+            index: null,
+            axes: [],
+            buttons: []
+        },
+        listeners: {
+            "onCreate.gamepadListener": "{that}.gamepadListener"
+        },
+        frequency: 100,
+        connectivityIntervalReference: null,
+        invokers: {
+            gamepadListener: {
+                funcName: "gamepad.gamepadNavigator.gamepadListener",
+                args: ["{that}"]
+            },
+            transformArrayToObject: {
+                funcName: "gamepad.gamepadNavigator.arrayToObject",
+                args: ["{arguments}.0"]
+            }
+        }
+    });
+
+    /**
+     *
+     * Listens to the connected and disconnected events of the gamepad and updates the
+     * gamepad navigator component's model according to the state of the gamepad.
+     *
+     * @param {Object} that - The gamepad navigator component.
+     *
+     */
+    gamepad.gamepadNavigator.gamepadListener = function (that) {
+        // When a gamepad is connected
+        window.addEventListener("gamepadconnected", function () {
+            // Store the gamepad info if no other gamepad is already connected.
+            if (!fluid.isValue(that.model.index)) {
+                // Scan the state of gamepad frequently.
+                that.options.connectivityIntervalReference = setInterval(function () {
+                    // Catch the current state of gamepad.
+                    var connectedGamepad = navigator.getGamepads()[0];
+
+                    // Initiate the gamepad navigator model transaction.
+                    var modelUpdateTransaction = that.applier.initiate();
+
+                    modelUpdateTransaction.fireChangeRequest({ path: "index", value: connectedGamepad.index });
```

Comment:
What happens if you have more than one controller?  I guess your code ignores the second one?

Reply:
I kept it on purpose as I thought that we don't want any second controller to interfere with the primary controller.

Reply:
Sorry, just saw your other comment on this approach. I'll reply to it shortly, after reading it.

Reply:
Thanks.  I think that supporting multiple controllers is really helpful, and it shouldn't be too hard IMO.